src/NzbDrone.Core/ImportLists/AniList/List/AniListImport.cs

```diff
@@ -0,0 +1,152 @@
+                    pageIndex = pageInfo.CurrentPage + 1; // increment using the returned server index for the current page
+                }
+                while (hasNextPage);
+
+                _importListStatusService.RecordSuccess(Definition.Id);
+            }
+            catch (WebException webException)
+            {
+                if (webException.Status == WebExceptionStatus.NameResolutionFailure ||
+                    webException.Status == WebExceptionStatus.ConnectFailure)
+                {
+                    _importListStatusService.RecordConnectionFailure(Definition.Id);
+                }
+                else
+                {
+                    _importListStatusService.RecordFailure(Definition.Id);
+                }
+
+                if (webException.Message.Contains("502") || webException.Message.Contains("503") ||
+                    webException.Message.Contains("timed out"))
+                {
+                    _logger.Warn("{0} server is currently unavailable. {1} {2}", this, url, webException.Message);
+                }
+                else
+                {
+                    _logger.Warn("{0} {1} {2}", this, url, webException.Message);
+                }
+            }
+            catch (TooManyRequestsException ex)
+            {
+                if (ex.RetryAfter != TimeSpan.Zero)
+                {
+                    _importListStatusService.RecordFailure(Definition.Id, ex.RetryAfter);
+                }
+                else
+                {
+                    _importListStatusService.RecordFailure(Definition.Id, TimeSpan.FromHours(1));
+                }
+
+                _logger.Warn("API Request Limit reached for {0}", this);
+            }
+            catch (HttpException ex)
+            {
+                _importListStatusService.RecordFailure(Definition.Id);
+                _logger.Warn("{0} {1}", this, ex.Message);
+            }
+            catch (RequestLimitReachedException)
+            {
+                _importListStatusService.RecordFailure(Definition.Id, TimeSpan.FromHours(1));
+                _logger.Warn("API Request Limit reached for {0}", this);
+            }
+            catch (CloudFlareCaptchaException ex)
+            {
+                _importListStatusService.RecordFailure(Definition.Id);
+                ex.WithData("FeedUrl", url);
+                if (ex.IsExpired)
+                {
+                    _logger.Error(ex, "Expired CAPTCHA token for {0}, please refresh in import list settings.", this);
+                }
+                else
+                {
+                    _logger.Error(ex, "CAPTCHA token required for {0}, check import list settings.", this);
+                }
+            }
+            catch (ImportListException ex)
+            {
+                _importListStatusService.RecordFailure(Definition.Id);
+                _logger.Warn(ex, "{0}", url);
```

Comment:
This is copied from the base class ( HttpImportListBase ) as-is, because the FetchItems method could not be used.

Please confirm. And if confirmed, please suggest a message. Whatever is decided here should probably be made to the base class for consistency.